### PR TITLE
ESP32-S2 platform support

### PR DESCRIPTION
**Pull request by systemofapwne**
This PR adresses two issues in order to make this package compatible with the ESP32-S2 platform.

- Hard-coded initializer for 'encoders' array will fail on platforms, where MAX_ESP32_ENCODERS is not 8:
```
Fixed bug, where the encoders array was hard-coded initialized to be 8 times NULL.
When running on a platform like ESP32-S2, where MAX_ESP32_ENCODERS is 4, the former limit caused a compile-error.
Changed initializer-list to { NULL } to cover arbitrary amount of encoders
```
- Compile error for ESP32-S2 due to variable-refactoring on esp32-arduino framework
```
Fixed compile error on ESP32-S2 platform as reported here https://github.com/madhephaestus/ESP32Encoder/issues/51
```

### CODE REVIEW

**Comment by madhephaestus** [comment]
good call, although you need to make it { NULL, } (note the trailing "," after null, that forces the compiler to fill the rest of the array as that value. In your code the first element will be null, but the rest of the values will be indeterminant. Push that change and ill merge this :) Thank You!

**Comment by systemofapwne** [comment]
> good call, although you need to make it { NULL, } (note the trailing "," after null) that forces the compiler to fill the rest of the array as that value. In your code the first element will be null, but the rest of the values will be indeterminant. Push that change and ill merge this :) Thank You!

I can surely do this later, however this might not be necessary (but a good practice).
To my knowledge and also testing with gnu c++ compiler, even an empty initializer list {} should be sufficient, to init all elements to NULL. Writing { NULL } just makes it clear about the value and { NULL, } more obvious about filling all elements (even thoug this happens for {}, { NULL } and { NULL, }).

**Comment by systemofapwne** [comment]
Added #ifdef guards to fully support ESP32-S2 platform (https://github.com/madhephaestus/ESP32Encoder/issues/51)

File: src/ESP32Encoder.cpp [hunk 1]
```diff
@@ -36,6 +34,13 @@ ESP32Encoder::~ESP32Encoder() {
  * and pass this information together with the event type
  * the main program using a queue.
  */
+#ifdef CONFIG_IDF_TARGET_ESP32S2
+	#define COUNTER_H_LIM cnt_thr_h_lim_lat_un
+	#define COUNTER_L_LIM cnt_thr_l_lim_lat_un
+#else
+	#define COUNTER_H_LIM h_lim_lat
+	#define COUNTER_L_LIM counter_l_lim
```
Comment (ptabor) [inline src/ESP32Encoder.cpp:42]: This should be `l_lim_lat`

[Tested this patch locally as I have encountered the same problem]

Comment (systemofapwne) [reply src/ESP32Encoder.cpp]: You are right! Shame on me, not testing the commit before pushing. I have fixed the typo.

**Review by ptabor — APPROVED** [review]

**Comment by madhephaestus** [comment]
Go team ESP32!
